Make-a-full-stack-change.md

```diff
@@ -0,0 +1,431 @@
+## Table of Contents
```

Comment:
Change title to reflect that this is not a guide to how to make a full-stack change. Rather, it is designed as an active exploration to accompany the codebase overview wiki page